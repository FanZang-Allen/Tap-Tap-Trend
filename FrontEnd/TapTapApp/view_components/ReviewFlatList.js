import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  ScrollView,
  SafeAreaView,
  FlatList,
  Image,
  RefreshControl,
} from 'react-native';

const ReviewFlatList = ({navigation, gotData, refreshing, onRefresh}) => {
  const Item = ({
    name,
    time,
    avatarUrl,
    rating,
    content,
    play_time,
    device,
  }) => (
    <View style={styles.item}>
      <View style={styles.header_container}>
        <View style={styles.avatar_container}>
          <Image
            source={{
              uri: avatarUrl,
            }}
            style={styles.avatar_image}
            resizeMode="contain"
          />
        </View>
        <Text style={styles.user_name}>{name}</Text>
        <View style={styles.rating_container}>
          <Image
            source={require('../assets/RatingStar.png')}
            style={styles.rating_image}
            resizeMode="contain"
          />
          <Text style={styles.rating_str}>{rating}</Text>
        </View>
        <Text style={styles.detial_str}>{time}</Text>
      </View>
      <View style={styles.content_container}>
        <ScrollView style={styles.content_scrollView}>
          <Text style={styles.content_text}>{content}</Text>
        </ScrollView>
      </View>
      <View style={styles.detail_container}>
        <Text style={styles.detial_str}>{play_time}</Text>
        <Text style={styles.detial_str}>{device}</Text>
      </View>
    </View>
  );

  const renderItem = ({item}) => (
    <Item
      name={item.reviewer}
      time={item.review_time}
      avatarUrl={item.avatar_url}
      rating={item.rating}
      content={item.content}
      play_time={item.play_time === undefined ? 'Played 0 min' : item.play_time}
      device={item.device === undefined ? 'Unknown Device' : item.device}
    />
  );
  function RenderList(data) {
    return (
      <SafeAreaView style={styles.container}>
        <FlatList
          data={data}
          renderItem={renderItem}
          keyExtractor={item => item.id}
          refreshControl={
            <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
          }
        />
      </SafeAreaView>
    );
  }
  function RenderFailed() {
    return (
      <View style={styles.background}>
        <Text style={styles.provider_name}>Not Available</Text>
      </View>
    );
  }
  function RenderLoading() {
    return (
      <View style={styles.background}>
        <Text style={styles.provider_name}>Loading</Text>
      </View>
    );
  }
  if (gotData === null) {
    return RenderLoading();
  } else if (gotData === undefined) {
    return RenderFailed();
  } else {
    return RenderList(gotData);
  }
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  background: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 20,
    paddingBottom: 10,
  },
  item: {
    backgroundColor: '#ffffff',
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    borderRadius: 15,
  },
  header_container: {
    height: '30%',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  content_container: {
    height: '60%',
    flex: 1,
  },
  content_scrollView: {
    marginLeft: 17,
    marginTop: 5,
  },
  content_text: {
    fontSize: 14,
    color: 'black',
    fontWeight: 'bold',
  },
  detail_container: {
    height: '10%',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginLeft: 15,
  },
  detial_str: {
    flex: 1,
    fontSize: 14,
    color: '#9B9B9B',
  },
  avatar_container: {
    width: '20%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  rating_container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    marginLeft: 5,
  },
  avatar_image: {
    flex: 1,
    width: 35,
    height: 35,
  },
  rating_image: {
    width: 20,
    height: 20,
  },
  user_name: {
    fontSize: 16,
    color: 'black',
    fontWeight: 'bold',
  },
  rating_str: {
    flex: 4,
    fontSize: 14,
    color: '#9076FB',
  },
  provider_name: {
    flex: 4,
    fontSize: 14,
    color: '#9B9B9B',
  },
});

export default ReviewFlatList;
