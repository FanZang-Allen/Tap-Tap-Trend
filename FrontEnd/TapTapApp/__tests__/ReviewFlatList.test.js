import 'react-native';
import React from 'react';
import ReviewFlatList from '../view_components/ReviewFlatList';
import renderer from 'react-test-renderer';

let testData = [
  {
    avatar_url:
      'https://img3.tapimg.com/avatars/etag/FhWKmP9njgQkVuntFY9xQpxIgfvg.png?imageMogr2/auto-orient/strip/thumbnail/!300x300r/gravity/Center/crop/300x300/format/jpg/interlace/1/quality/40',
    content: 'looks great',
    device: ' Vivo V1955A ',
    id: 0,
    link: 'https://www.taptap.io/review/2148582585',
    rating: '5.0',
    review_time: '22 hr ago',
    reviewer: '康太郎',
  },
];

test('Game Flat List Rendered Page', () => {
  const snap = renderer
    .create(
      <ReviewFlatList
        gotData={testData}
        refreshing={false}
        onRefresh={() => console.log('This screen not suppport refresh')}
      />,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
