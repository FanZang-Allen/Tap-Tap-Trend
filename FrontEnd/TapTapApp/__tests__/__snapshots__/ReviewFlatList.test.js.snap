// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Flat List Rendered Page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "avatar_url": "https://img3.tapimg.com/avatars/etag/FhWKmP9njgQkVuntFY9xQpxIgfvg.png?imageMogr2/auto-orient/strip/thumbnail/!300x300r/gravity/Center/crop/300x300/format/jpg/interlace/1/quality/40",
          "content": "looks great",
          "device": " Vivo V1955A ",
          "id": 0,
          "link": "https://www.taptap.io/review/2148582585",
          "rating": "5.0",
          "review_time": "22 hr ago",
          "reviewer": "康太郎",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 15,
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-evenly",
              "marginHorizontal": 16,
              "marginVertical": 8,
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "height": "30%",
                "justifyContent": "flex-start",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "width": "20%",
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "uri": "https://img3.tapimg.com/avatars/etag/FhWKmP9njgQkVuntFY9xQpxIgfvg.png?imageMogr2/auto-orient/strip/thumbnail/!300x300r/gravity/Center/crop/300x300/format/jpg/interlace/1/quality/40",
                  }
                }
                style={
                  Object {
                    "flex": 1,
                    "height": 35,
                    "width": 35,
                  }
                }
              />
            </View>
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              康太郎
            </Text>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginLeft": 5,
                }
              }
            >
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "height": 20,
                    "width": 20,
                  }
                }
              />
              <Text
                style={
                  Object {
                    "color": "#9076FB",
                    "flex": 4,
                    "fontSize": 14,
                  }
                }
              >
                5.0
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#9B9B9B",
                  "flex": 1,
                  "fontSize": 14,
                }
              }
            >
              22 hr ago
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "height": "60%",
              }
            }
          >
            <RCTScrollView
              style={
                Object {
                  "marginLeft": 17,
                  "marginTop": 5,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 14,
                      "fontWeight": "bold",
                    }
                  }
                >
                  looks great
                </Text>
              </View>
            </RCTScrollView>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "height": "10%",
                "justifyContent": "flex-start",
                "marginLeft": 15,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#9B9B9B",
                  "flex": 1,
                  "fontSize": 14,
                }
              }
            >
              Played 0 min
            </Text>
            <Text
              style={
                Object {
                  "color": "#9B9B9B",
                  "flex": 1,
                  "fontSize": 14,
                }
              }
            >
               Vivo V1955A 
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
